subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/inputs/undeclared/GroovyPluginImplementation.groovy

```diff
@@ -89,14 +91,16 @@ trait GroovyPluginImplementation {
                     }
                     cl("apply")
 
-                    project.tasks.register("thing") { t ->
-                        t.doLast {
-                            value = ${read.groovyExpression}
-                            println("task = " + value)
+                    if (!project.tasks.names.contains("thing")) {
```

Comment:
I'm not sure if introducing discrepancies into plugin implementations is a good idea. Maybe we can get away with an ad hoc implementation of the template in the test itself? Then you can drop all unrelated checks for INSTANCE, etc.